MetadataExtractor/IO/IndexedCapturingReader.cs

```diff
@@ -51,6 +51,40 @@ public IndexedCapturingReader([NotNull] Stream stream, int chunkLength = Default
 
             _chunkLength = chunkLength;
             _stream = stream ?? throw new ArgumentNullException(nameof(stream));
+
+            try
+            {
+                // For some reason, FileStreams are faster in contiguous mode. Since this is such a a commont case, we
+                // specifically check for it.
```

Comment:
I think the underlying OS pre-fetches data. Many read operations are sequential. In fact, when you create a `FileStream` there's a ctor overload that lets you specify `Sequential` vs `RandomAccess` (IIRC) which presumably hints the platform of how the reader will behave. For the places that we create `FileStream` we could experiment there to see if it matters.

Reply:
Right, so what I didn't understand is why the performance would have changed: in the case where it previously pre-loaded the whole file by reading sequentially (on FileStreams), we still do that. Likewise, the cost of the lookup was all but eliminated by memoizing the last looked up chunk (only something like 5% of calls ended up requesting a chunk other than the last one it returned). But the new code is still ~50% slower at running the test-suite.

Alas I am developing on a Mac, and the Xamarin-based toolchain here doesn't have a robust profiler (nor do I have VS Ultimate (or whatever it's called) on my Windows machine). If you do have access to a profiler, I'd be very interested to know what you see.